target_game.py

```diff
@@ -23,23 +26,119 @@ def move_target_at_random_pos(target):
     new_pos = (pos_x, pos_y)
     target.setPos(new_pos)
     
+
+# Now I create a function that will draw 3 targets - each for each level of the game
+# On these targets I want to show where the player clicked (scored) on each level
+def draw_overview_target(target_x, target_y, mouse_clicks):
+    target = ImageStim(win, 'target.png', size = (420, 420), pos=(target_x, target_y))
+    target.draw()
+    
+    for target_hit_pos in mouse_clicks:
+        circle = Circle(win, radius=5, fillColor='yellow', lineColor='yellow')
+        circle.setPos((target_hit_pos['mouse_x']+target_x, target_hit_pos['mouse_y']+target_y))
+        circle.draw()
+
+# Here I create a function for the intro, which allows me to explain the game to the participant
+def show_intro(win):
+    start = TextStim(
+        win,
+        text = "Hi there, player! This is you! Let's play!",
+        height = 35,
+        color = (0.2,0.2,0.8),
+        pos = (250,230),
+    )
+    
+    player = ImageStim(
+        win,
+        'creature.jpg',
+        pos = (0,-100),
+        size=(420,420),
+    )
+    
+    intro_text = (
+        'In this game you should shoot at the target with your mouse. '
+        + 'The target moves rapidly to a new position each time you shoot at it.' 
+        + 'There are 3 levels each faster than the previous one. Good luck!'
+    )
+    
+    instructions = TextStim(
+        win,
+        text = intro_text,
+        height = 35,
+        color = (1, 0.2, 0.6),
+        pos = (250, 0),
+    )
+    
+    start.draw()
+    player.draw()
+    win.flip()
+    wait(3)
+    instructions.draw()
+    win.flip()
+    wait(8)           

```

Comment:
I think this last bit belongs in your game loop and not in the function...

Reply:
In general, I would execute all functions in the final loop and not in the function itself but maybe I am wrong (I don't know so much about psychopy).

Reply:
I think either way it works fine. Since I am using it only once I can indeed take it out of the function and put it in the loop. However, I thought the intro might be skipped if needed, and then it will be easier to put this part in the function, so it would not go if I disable the intro.